packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/jni/JavaScriptModuleObject.kt

```diff
@@ -0,0 +1,27 @@
+package expo.modules.kotlin.jni
+
+import com.facebook.jni.HybridData
+import com.facebook.react.bridge.ReadableNativeArray
+import expo.modules.kotlin.ModuleHolder
+import java.lang.ref.WeakReference
+
+class JavaScriptModuleObject(moduleHolder: ModuleHolder) {
+  // Has to be called "mHybridData" - fbjni uses it via reflection
+  private val mHybridData = initHybrid()
+  private val moduleHolderRef = WeakReference(moduleHolder)
+
+  @Suppress("KotlinJniMissingFunction")
+  private external fun initHybrid(): HybridData
+
+  @Suppress("KotlinJniMissingFunction")
+  external fun registerSyncFunction(name: String, args: Int)
```

Comment:
Looks like we don't register sync functions anywhere, I don't see any call site of this yet. Where is it supposed to be called? 🤔

Reply:
Yes, you're right. Sorry for that. I did add registration in the module holder.